### ISSUE TITLE
Use the Goertzel algorithm to compute the DFT.

### ISSUE THREAD

**Issue by shiangsir** [rudolfbyker/lpsd#2]
@tobin has another version of the LPSD implementation in Matlab:

https://github.com/tobin/lpsd/tree/goertzel

He uses the Goertzel algorithm to compute the DFT in his Matlab code. I think this is far more efficient in doing the LPSD. You may try it in your Python version.  I am doing a project and want to use the LPSD method in the Python environment as you did. Thank you for sharing your Python code. It helps us a lot. Thanks!

**Comment by rudolfbyker** [comment]
Hi there, thanks for the issue. If you have time to implement it, feel free to send me a pull request.

I ended up not using the LPSD for the project that I was working on. What I actually needed was a short-time transform with an optimal tradeoff between time-resolution and frequency-resolution for all frequencies. As soon as I stumbled upon the [Constant-Q transform implemented here](https://librosa.github.io/librosa/generated/librosa.core.cqt.html), I stopped considering the LPSD.
